### PR TITLE
feat(analysis): capture bounded private run-catalog generations

### PR DESCRIPTION
**Pull request by andreasronge**
Refs #1642 — implements **step 1 of the four-PR breakdown** frozen in [the design-decisions comment](https://github.com/andreasronge/ptc_runner/issues/1642#issuecomment-5445118504): the `RunCatalog` capture owner. Steps 2–4 remain; see **Remaining work** below.

> Current head `5d6921e`. The review of the first revision raised six findings, all fixed; every review thread is resolved and no question is outstanding. See **Review history** below.

## The frozen contract this implements

Per §1, §2, §4 and §5 of the design comment, as amended by [the accepted schema amendment](https://github.com/andreasronge/ptc_runner/issues/1642#issuecomment-5449287788):

- **Derived immutable snapshot, no sidecars, no persisted catalog file.** A generation is an in-memory capture owned by `RunCatalogSnapshot`, built from bounded probes: a `.ptcins` container's 16-byte header and 192-byte footer (two `pread`s, no frame decoded), and a canonical trace's first line plus the last complete line inside a bounded tail window.
- **Identity is verified, never trusted from a filename.** `sha256(stem) == footer.run_id_sha256` and `sha256(trace head trace_id) == footer.trace_id_sha256`; a disagreement isolates the row. The two probed trace events are additionally validated by `TraceEventValidation`, so a file the canonical reader rejects is never published as admissible.
- **The bytes read are the bytes committed to.** Each file is `lstat`ed before the open, the opened descriptor is `fstat`ed and compared before a byte is read, and the path is `lstat`ed again after — so neither a file replaced under its path nor one rewritten mid-probe is described as the file that was listed.
- **Generation digest.** `sha256("ptc-run-catalog-v1\0" <> sorted per-row commitments)`, where each commitment is `{run_ref, trace half, inspection half}` and each half is `{present, descriptor identity, sha256 of the exact bytes probed}`. Committing to *bytes* rather than to decoded values is load-bearing: a footer's stored `artifact_digest` is a claim a metadata-only probe never recomputes, so a commitment derived from it misses a rewrite of any other footer field. Halves that read bytes but did not decode — malformed, foreign-version — commit to them too; only a half that never reached any bytes carries an absent marker. Nothing re-lists after capture, so a growing root cannot alter an open generation.
- **Row schema (§2)** as specified, including the exclusions: no prompts, responses, generated source, capability payloads, diagnostics, prints, result values, evidence content, filesystem paths, or any field needing a whole-trace scan. Plus the two amended enum values, `correlation: "unavailable"` and `trace_present: "unreadable"`, for an entry whose probe failed on a half.
- **Failure matrix (§4).** Per-row isolation with path-free codes — `ambiguous_trace`, `malformed_metadata`, `unstable_entry`, `unsupported_schema`, `filename_run_mismatch`, `duplicate_run_identity` — and whole-operation refusals `source_unavailable` / `catalog_limit_exceeded`. A malformed *unselected* entry never blocks the generation. Every duplicate claimant is isolated; the primary reason follows existing precedence.
- **Limits (§5).** 4,096 directory entries, 1,024 stems, 65,536-byte head + 65,536-byte tail + 208 sealed edge bytes per file, 2,048-byte rows, 4,194,304 retained bytes, 15,000 ms capture deadline, 5,000 ms listing deadline. All may be lowered by a caller and none raised.

## Decisions the frozen comment left unstated

1. **`correlation: "unavailable"` and `trace_present: "unreadable"`** — values beyond the original enums, for an entry whose probe failed on a half. **Proposed on #1642 and accepted into the frozen contract.**
2. **Correlation-mismatch row reason** — reuses the existing `inspection_correlation_missing`, the code the matrix already assigns to that state on the selection side.
3. **Oversized-row reason** — the reduced row is isolated as `malformed_metadata`.
4. **Header/footer version disagreement is `malformed_metadata`, not `unsupported_schema`** — an internally inconsistent container is corruption, not a version this build cannot read.
5. **A tail `run-stopped` naming a different run is `malformed_metadata`** — the two lines disagree about whose file it is.
6. **A non-integer `schema_version` is `malformed_metadata`, not `unsupported_schema`** — it is not a version this build cannot read, it is an event the canonical reader rejects.
7. **Identity strings bounded to 256 bytes** (matching `TraceDirectoryAdmission`'s existing claim bound) **and timestamps to 64 bytes.** ISO 8601 admits an unbounded fractional-second field, so a parse alone is not a bound; the row byte bound cannot drop either field, so both are bounded at the probe.
8. **Extracted `BoundedCapture`.** The duplication gate flagged the owner-capture loop as a copy of `TraceSnapshot`'s; both now share one owner-scoped implementation, each mapping its outcome onto its own source codes.

## Verification

Toolchain had to be installed in this container (`mise install` → Erlang 29.0.3 / Elixir 1.20.2, per `mise.toml`).

| Command | Result on `5d6921e` |
|---|---|
| `mix compile --warnings-as-errors` | clean |
| `mix precommit` | passes; duplication `new=0` |
| `mix dialyzer --format short` | `Total errors: 0` |
| `MIX_ENV=dev mix docs --warnings-as-errors` | clean (it caught a moduledoc link to a hidden module) |
| `mix test` (catalog files) | 45 passed |
| GitHub CI | all checks green |

The full suite in this container reports seven **pre-existing environment failures** unrelated to this diff: re-running those files on clean `main` in the same container reproduces the identical seven. Four are `chmod`-based tests asserting `:eacces`, defeated by the container running as root; two need IPv6 (`:eafnosupport`); one REPL frontend test is in the same class. GitHub CI passes all of them, which confirms the diagnosis.

The repo's pre-push hook is not installed in this fresh clone (`core.hooksPath` unset), so each gate it invokes was run by hand.

## Review history

Six reproduced findings on the first revision, all fixed. Each reproduction is now a test.

**P1 — probed events were not validated as canonical trace events.** Projecting from a handful of hand-picked fields let a file the canonical reader rejects be published as complete and admissible. The two probed events now go through `TraceEventValidation`. `{:error, :unsupported_version}` still projects so the row can report its declared version; every other failure isolates the row.

**P1 — the generation digest was bound to decoded values.** Rewriting `record_count` in a footer changed the row while leaving `catalog_digest` identical, because the commitment was built from the footer's stored `artifact_digest`. Both halves now commit to the exact bytes probed plus descriptor identity.

**P2 — the opened descriptor was never `fstat`ed.** A path replaced between `lstat` and `open` and restored before the closing `lstat` could commit replacement bytes under the original identity. Fixed following `Handle.open/2`; this required dropping `:raw`, since a raw descriptor is process-local and cannot be `fstat`ed. I judged the window untestable without a `Process.sleep` or a test-only export; the maintainer showed otherwise in `5d6921e`, adding a nil-by-default `:file_probe_hook` on the same footing as the existing `:listing_hook` and a rendezvous-based regression that swaps the file before `open` and restores it before the closing `lstat`, so only the `fstat` can catch it.

**P2 — current-version header geometry was unvalidated.** A declared header size of 17 was accepted although the artifact reader rejects it. `Format.decode_header/1` now validates the whole header once the versions are recognised as this build's.

**P2 — duplicate detection ignored inspection-footer identity claims.** Traces state identities in plaintext and footers only as digests, so the two halves claimed into spaces that could never intersect, and copying one run's artifact over another's canonical name left the impersonated run admissible. The plaintext is now hashed to meet the footer.

**P2 — classifier inputs were only partially validated.** The partial check advertised `{:error, :invalid_catalog}` and then broke it by raising out of a projection. Every field the classifier reads is now validated.

## Remaining work

Steps 2–4 of the frozen breakdown are untouched by this PR:

2. **`analysis-catalog` capability + `private-run-catalog-v1` profile** — paging, the filter vocabulary plus `correlation`/`state`, digest-bound cursors reusing the `TraceLog` cursor helpers, `excluded_files` accounting, REPL/JSONL wiring. `RunCatalogSnapshot.rows/1` is the frozen list that surface will page over.
3. **Bounded multi-run selected capture** — `{:selected_canonical_set, dir, run_refs}` in `SelectedCanonicalSource` and both snapshots, repeated `--run` on `private-run-analysis-v1`, `max_selected_runs` (16), aggregate-limit enforcement, and the `duplicate_selected_run` / `selected_set_limit_exceeded` / isolated-run refusals.
4. **Docs, diagnostics, integration** — `docs/reference/repl.md`, `cli.md`, `docs/maintainers/trace-log-contract.md`, `debug-navigation.md`, stable code-catalog additions.

Because the catalog has no user-facing surface until step 2, this PR adds no user documentation; the contract lives in the three module docs. Acceptance-list coverage is correspondingly partial: catalog growth during paging, stale cursors, and bounded incremental opening are step 2/3 concerns. Covered here: duplicate identities across both halves, missing pairs, changed manifests, malformed entries selected and unselected, non-canonical events, descriptor replacement under the inventoried path, large catalogs, generation immutability under a growing root, digest stability and change, payload-content leak checks, and owner cleanup.

### CODE REVIEW

File: lib/ptc_runner/kernel/run_catalog.ex [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+  defp commitment(%{run_ref: run_ref, trace: trace, inspection: inspection}) do
+    {run_ref, trace_commitment(trace), inspection_commitment(inspection)}
+  end
+
+  defp trace_commitment(%{present: :probed, identity: identity, commitment: commitment}),
+    do: {:probed, identity, commitment}
+
+  defp trace_commitment(%{present: present}), do: {present}
+
+  defp inspection_commitment(%{present: :probed} = inspection),
```
Comment (andreasronge) [inline lib/ptc_runner/kernel/run_catalog.ex]: **[P1] Bind the generation digest to the exact footer bytes.** The row reads record count, identity hashes, and other metadata from the footer, but this commitment includes only file size and the footer's claimed artifact digest. That digest is not verified during metadata-only probing. I reproduced changing `inspection_record_count` from 11 to 12 while `catalog_digest` remained identical. Future cursors could therefore accept a different catalog result as the same generation. Commit to a digest of the exact opened header/footer bytes and descriptor identity, including malformed and foreign-version probes.

File: lib/ptc_runner/kernel/run_catalog_probe.ex [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+    case candidate |> :binary.split("\n", [:global]) |> Enum.reverse() do
+      [_trailing | terminated] -> Enum.find(terminated, &(&1 != "")) |> wrap_line()
+      [] -> :error
+    end
+  end
+
+  defp wrap_line(nil), do: :error
+  defp wrap_line(line), do: {:ok, line}
+
+  defp decode_started(line) do
```
Comment (andreasronge) [inline lib/ptc_runner/kernel/run_catalog_probe.ex]: **[P1] Validate the probed events as canonical trace events before projecting them.** The head/tail parser validates only a few selected fields. After deleting `trace_id` from `run-stopped`, current-main `TraceEventValidation` returned `malformed_source`, while this catalog reported the run as complete and admissible. Missing schema, sequence, timestamp/data structure, current terminal usage fields, mismatched trace identity, and invalid result hashes have similar paths. Please factor out bounded canonical validation for the probed events.

File: lib/ptc_runner/kernel/run_catalog_probe.ex [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+          _moved ->
+            %{present: :unstable}
+        end
+
+      :error ->
+        %{present: :malformed}
+    end
+  end
+
+  defp read_edges(path, size) do
```
Comment (andreasronge) [inline lib/ptc_runner/kernel/run_catalog_probe.ex]: **[P2] Verify that the opened descriptor is the inventoried file.** The probe lstats the path, opens it, reads it, and lstats the path again, but never fstats the opened descriptor. A path can be replaced between lstat and open and restored before the final lstat, causing replacement bytes to be committed with the original file identity. Fstat the opened descriptor and compare it to the initial stat, following `InspectionArtifact.Handle.open/2`.

File: lib/ptc_runner/kernel/run_catalog_probe.ex [hunk 4]
```diff
@@ -0,0 +1,511 @@
...
+    header = binary_part(head, 0, Format.header_size())
+
+    footer_bytes =
+      binary_part(
+        tail_bytes,
+        byte_size(tail_bytes) - Format.footer_size(),
+        Format.footer_size()
+      )
+
+    with {:ok, header_versions} <- Format.decode_header_versions(header),
```
Comment (andreasronge) [inline lib/ptc_runner/kernel/run_catalog_probe.ex:586]: **[P2] Reject invalid current-version header geometry.** `decode_header_versions/1` ignores the declared header size, and the current-version path never calls `Format.decode_header/1`. I changed the header-size field from 16 to 17 and the catalog still returned a paired, admissible row, although the normal artifact reader rejects it as malformed. Once versions are recognized as current, validate the complete current header before accepting the footer projection.

**Review by andreasronge — COMMENTED** [review]
Reviewed against updated `origin/main` at `1ad966ea`, not only the PR's original base. The synthetic merge was clean; the 38 new catalog tests and 50 impacted trace/snapshot tests passed. The inline comments below capture six reproduced or directly reachable correctness issues.

Contract drift to settle before step 2: the frozen #1642 schema lists four correlation values and three trace-presence values, while this PR adds `"unavailable"` and `"unreadable"`. Those may be sensible, but they should be added to the issue's frozen contract rather than living only in the PR body.

GitHub CI is green, but it does not cover the reproduced cases in the inline findings.

File: lib/ptc_runner/kernel/run_catalog.ex [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+  defp duplicated_identities(probes) do
+    probes
+    |> Enum.flat_map(&claimed_identities/1)
+    |> Enum.group_by(&elem(&1, 0), &elem(&1, 1))
+    |> Enum.filter(fn {_identity, refs} -> refs |> Enum.uniq() |> length() > 1 end)
+    |> Enum.flat_map(fn {_identity, refs} -> refs end)
+    |> MapSet.new()
+  end
+
+  defp claimed_identities(%{run_ref: run_ref, trace: %{present: :probed, head: head}}),
```
Comment (andreasronge) [inline lib/ptc_runner/kernel/run_catalog.ex]: **[P2] Include inspection-footer identity claims in duplicate detection.** This graph considers only plaintext identities from trace heads. It ignores `run_id_sha256` and `trace_id_sha256` claimed by inspection footers, despite the frozen failure matrix requiring all duplicate claimants to be isolated. Copying run A's `.ptcins` over run B's canonical inspection name left A admissible and isolated only B as a filename mismatch. Add the hashed inspection claims so every claimant is isolated.

File: lib/ptc_runner/kernel/run_catalog.ex [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+  # honest answer to a probe the classifier cannot read: half a row would
+  # publish a state nothing observed.
+  defp valid_probe?(%{run_ref: run_ref, trace: trace, inspection: inspection}) do
+    is_binary(run_ref) and run_ref != "" and valid_trace_probe?(trace) and
+      valid_inspection_probe?(inspection)
+  end
+
+  defp valid_probe?(_probe), do: false
+
+  defp valid_trace_probe?(%{present: :probed, head: head, tail: tail} = trace) do
```
Comment (andreasronge) [inline lib/ptc_runner/kernel/run_catalog.ex:374]: **[P2] Finish validating classifier inputs before building rows.** `valid_trace_probe?/1` checks only a subset of fields subsequently read by the classifier. A probe with `head.timestamp: 42` and a non-nil tail passes validation and then raises inside `DateTime.from_iso8601/1`, contradicting `generation/2`'s `{:error, :invalid_catalog}` contract. Validate the byte/count/commitment, head timestamp/schema, and complete tail field types here.

File: lib/ptc_runner/kernel/run_catalog_probe.ex (hunk 2 above)
Comment (andreasronge) [reply lib/ptc_runner/kernel/run_catalog_probe.ex]: Fixed in 484ebaf. Rather than write a parallel validator I hand the two probed events to `TraceEventValidation.validate/1` — it already owns per-event shape, run/trace identity agreement, sequence ordering and run lifecycle, and validating exactly the two events the row is built from keeps it bounded. The projection now happens only after it returns.

One deliberate exception: `{:error, :unsupported_version}` still projects, so the row can report the version it declares rather than being called corrupt. Every other failure isolates the row with nothing projected.

Two consequences worth flagging. A `schema_version` that is not an integer is now `malformed_metadata` rather than `unsupported_schema` — it is not a version this build cannot read, it is an event the canonical reader rejects — and I updated that test accordingly. And the tail is only validated alongside the head when it is a `run-stopped`; for a run still being written I validate the head alone, because an arbitrary mid-run last line would drag `advance_evaluation_lifecycle` into judging events the probe never read.

Your reproduction is now a test: deleting `trace_id` from `run-stopped` isolates the row. I added a second for the result-hash-contradicting-outcome path you mentioned.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/ptc_runner/kernel/run_catalog.ex (hunk 1 above)
Comment (andreasronge) [reply lib/ptc_runner/kernel/run_catalog.ex]: Fixed in 484ebaf, and you identified the root cause exactly: I was committing to decoded values, and a footer's `artifact_digest` is a stored claim a metadata-only probe never recomputes — so editing `record_count` beside it changed the row while leaving my commitment alone.

Both halves now commit to `{present, descriptor identity, sha256 of the exact bytes probed}` — the trace's head and tail windows, the container's header and footer — which collapsed the two commitment functions into one shared `half_commitment/1`. Malformed and foreign-version probes carry it too, so a half that reached any bytes always commits to them; only a half that never read any (absent, or identity moved) carries neither.

Your reproduction is now a test: rewriting `record_count` in place changes `catalog_digest` and the row together.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/ptc_runner/kernel/run_catalog_probe.ex (hunk 3 above)
Comment (andreasronge) [reply lib/ptc_runner/kernel/run_catalog_probe.ex]: Fixed in 484ebaf, following `Handle.open/2` as suggested: `lstat` the path, open, `fstat` the descriptor and compare before a single byte is read, then `lstat` again after. A mismatch reports `:unstable` rather than committing the bytes.

This required dropping `:raw` from the open — a raw descriptor is process-local and cannot be `fstat`ed — so the probe now uses a file-server descriptor like `Handle` does. It is two `pread`s per file, opened and closed within the probe, so no handles are retained.

I did not add a test for this one. The window is only reachable by winning a race against the probe, and the alternatives were a test-only export on the probe or a `Process.sleep`, neither of which belongs here. What the existing tests do prove is the negative: if the comparison were wrong in the other direction every probe would report `:unstable`, and all 44 pass. Happy to add instrumentation if you would rather have it covered explicitly.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/ptc_runner/kernel/run_catalog_probe.ex (hunk 4 above)
Comment (andreasronge) [reply lib/ptc_runner/kernel/run_catalog_probe.ex:586]: Fixed in 484ebaf. `current_inspection_row/4` now starts with `Format.decode_header/1` on the header bytes, so once the versions are recognised as this build's, the whole header is validated the way the artifact reader validates it — including the declared size — before any footer field is projected.

`decode_header_versions/1` keeps ignoring the size deliberately: its whole job is to read the two version fields of a container whose layout this build may not know, and a foreign version may legitimately declare a different header size. It is now only the gate that decides *which* validation applies, never the validation itself.

Your reproduction is now a test: a header-size field of 17 isolates the row as `malformed_metadata`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/ptc_runner/kernel/run_catalog.ex (hunk 5 above)
Comment (andreasronge) [reply lib/ptc_runner/kernel/run_catalog.ex]: Fixed in 484ebaf. Both halves now claim into one space: a trace states its identities in plaintext and a footer states them only as digests, so I hash the plaintext to meet the footer rather than trusting the footer to name itself. Previously the two halves claimed into spaces that could never intersect, which is why the impersonated run stayed admissible.

One thing I want to flag rather than quietly decide. In your reproduction both entries are now isolated, but they report different reasons: A gets `duplicate_run_identity` (its only defect), while B gets `filename_run_mismatch`, which outranks it in my reason ordering. B genuinely has both defects, and the mismatch is the more specific and more actionable one — its bytes are not the run its name claims. Given the stem is verified against the footer hash, a copied artifact essentially always carries a filename mismatch too, so forcing both to `duplicate_run_identity` would mean suppressing a truer reason.

I read the frozen matrix's requirement as "all claimants isolated", which now holds, and the test asserts both the isolation and each specific reason. If you intended the reason itself to be uniform across claimants, say so and I will reorder `duplicate_run_identity` above `filename_run_mismatch`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/ptc_runner/kernel/run_catalog.ex (hunk 6 above)
Comment (andreasronge) [reply lib/ptc_runner/kernel/run_catalog.ex:374]: Fixed in 484ebaf. You are right that the partial check was worse than none — it advertised `{:error, :invalid_catalog}` and then broke that contract by raising out of a projection instead.

`valid_trace_probe?/1` now checks the byte count, descriptor identity, commitment and source kind, the head's `run_id`/`trace_id`/`labels`/`timestamp`/`schema_version`, and every tail field type; `valid_inspection_probe?/1` does the same for both its probed and foreign-version shapes. The rule I applied is that anything the classifier later reads is checked here, without exception.

Your `head.timestamp: 42` case plus seven siblings (head schema, all four tail fields, bytes, commitment) are now a table-driven test asserting `{:error, :invalid_catalog}` for each.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/ptc_runner/kernel/run_catalog_probe.ex (hunk 3 above)
Comment (andreasronge) [reply lib/ptc_runner/kernel/run_catalog_probe.ex]: Closed in `5d6921e2`. The probe now exposes deterministic test synchronization immediately before open and immediately after open but before `fstat`. The regression replaces the inventoried path, waits until the replacement descriptor is open, restores the original path, and then resumes. With the descriptor comparison temporarily removed, the row incorrectly becomes `admissible`; with the fix present, it is isolated as `unstable_entry`. No sleep or probabilistic retry is involved. The 101 focused tests and full repository pre-push gate pass.

File: lib/ptc_runner/kernel/run_catalog.ex (hunk 5 above)
Comment (andreasronge) [reply lib/ptc_runner/kernel/run_catalog.ex]: Decision accepted and frozen in [the #1642 contract amendment](https://github.com/andreasronge/ptc_runner/issues/1642#issuecomment-5449287788). Every duplicate claimant must be isolated, while the single primary reason follows the existing precedence: the correctly named claimant reports `duplicate_run_identity`; the copied artifact under the wrong canonical stem reports the more specific `filename_run_mismatch`. The current implementation and regression are therefore the intended contract.

**Comment by andreasronge** [comment]
## Merge-time cleanup completed

- `5d6921e2` adds deterministic coverage for replacement between initial `lstat` and descriptor verification. The counterfactual test publishes the substitution as `admissible` without the `fstat` comparison and isolates it as `unstable_entry` with the fix.
- The #1642 amendment now explicitly freezes `correlation: "unavailable"`, `trace_present: "unreadable"`, and primary-reason precedence for duplicate claimants.
- Both remaining review threads are resolved.
- The PR body now references the current commits and verification results.
- #1642 is an implementation-ready epic; #1681, #1682, and #1683 track steps 2–4.
- Local focused tests (101), `mix precommit`, the full repository pre-push gate, and GitHub CI all pass.

Process note: this follow-up was manually verified by Codex without the independent-review skill, following the instruction on this PR task. The repository-prescribed independent cumulative review is therefore not claimed here, and I have not applied `ready-to-merge` on that basis.

File: lib/ptc_runner/kernel/run_catalog_probe.ex (hunk 3 above)
Comment (andreasronge) [reply lib/ptc_runner/kernel/run_catalog_probe.ex]: Reviewed `5d6921e2` and it is better than what I declined to write — thank you. Two things I had wrong: I treated "no deterministic test" as a property of the race, when it was a property of the seam I was unwilling to add; and I assumed the seam would have to be a test-only export, when a nil-by-default hook validated for arity is exactly the shape `:listing_hook` already established, so it adds no reachable production surface.

The rendezvous is the part I could not see: swapping before `open` and restoring before the closing `lstat` means both path-based stats observe the original while the descriptor is the replacement, so the `fstat` is the only thing standing between that and a committed row. Verifying it by removing the comparison and watching the row go `admissible` is the check that makes it a real regression rather than a passing assertion.

Confirmed on my side at `5d6921e`: 45 catalog tests pass, `mix precommit` clean with duplication `new=0`, `mix dialyzer` 0 errors.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/ptc_runner/kernel/run_catalog.ex (hunk 5 above)
Comment (andreasronge) [reply lib/ptc_runner/kernel/run_catalog.ex]: Understood — isolation is the invariant, the primary reason follows existing precedence, and the current implementation and regression are the intended contract. Nothing to change; resolving.

---
_Generated by [Claude Code](https://claude.ai/code)_
